app/routes/public/sessions/list.js

```diff
@@ -16,169 +16,97 @@ export default Route.extend({
-            ]
+            name : 'starts-at',
+            op   : 'ge',
+            val  : moment().startOf('day').toISOString()
+          },
+          {
+            name : 'starts-at',
+            op   : 'lt',
+            val  : moment().endOf('day').toISOString()
           }
         ]
       });
-    } else if (params.session_status === 'month') {
-      sessions = await this.store.query('session', {
-        filter: [
+
+    } else if (params.session_status === 'week') {
+
+      filterOptions.push({
+        and: [
           {
-            and: [
-              {
-                name : 'event',
-                op   : 'has',
-                val  : {
-                  name : 'identifier',
-                  op   : 'eq',
-                  val  : eventDetails.id
-                }
-              },
-              {
-                name : 'starts-at',
-                op   : 'ge',
-                val  : moment().startOf('month').toISOString()
-              },
-              {
-                name : 'starts-at',
-                op   : 'lt',
-                val  : moment().add('month').toISOString()
-              },
-              {
-                or: [
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'confirmed'
-                  },
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'accepted'
-                  }
-                ]
-              }
-            ]
+            name : 'starts-at',
+            op   : 'ge',
+            val  : moment().startOf('week').toISOString()
+          },
+          {
+            name : 'starts-at',
+            op   : 'lt',
+            val  : moment().endOf('week').toISOString()
           }
         ]
       });
-    } else {
-      sessions = await this.store.query('session', {
-        filter: [
+
+    } else if (params.session_status === 'month') {
+
+      filterOptions.push({
+        and: [
           {
-            and: [
-              {
-                name : 'event',
-                op   : 'has',
-                val  : {
-                  name : 'identifier',
-                  op   : 'eq',
-                  val  : eventDetails.id
-                }
-              },
-              {
-                or: [
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'confirmed'
-                  },
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'accepted'
-                  }
-                ]
-              }
-            ]
+            name : 'starts-at',
+            op   : 'ge',
+            val  : moment().startOf('month').toISOString()
+          },
+          {
+            name : 'starts-at',
+            op   : 'lt',
+            val  : moment().endOf('month').toISOString()
```

Comment:
Still duplicated thrice

Reply:
How can I filter month, week, day better than this?